src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,241 @@
+                                createCarRequest, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet generatedKeys = createCarStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                car.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create car: " + car, e);
+        }
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getRequest = "SELECT c.id, c.model, "
+                + "m.id AS manufacturer_id, m.name, m.country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE and c.id = ?;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement =
+                        connection.prepareStatement(getRequest)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+            return Optional.of(car);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllFromCarsTableRequest = "SELECT *"
+                + "FROM cars c JOIN manufacturers m "
+                + "ON m.id = c.manufacturer_id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement =
+                        connection.prepareStatement(getAllFromCarsTableRequest)) {
+
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars "
+                    + "from cars table. ", e);
+        }
+        cars.forEach(c -> c.setDrivers(getDriversForCar(c.getId())));
+        return cars;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverRequest = "SELECT c.id, "
+                + "c.model AS model, "
+                + "m.id AS manufacturer_id, "
+                + "m.name AS name, "
+                + "m.country AS country "
```

Comment:
```suggestion
                + "c.model, "
                + "m.id AS manufacturer_id, "
                + "m.name, "
                + "m.country "
```
in my opinion it may be so